sympy/functions/elementary/complexes.py

```diff
@@ -662,7 +662,9 @@ def _eval_nseries(self, x, n, logx, cdir=0):
         if direction.has(log(x)):
             direction = direction.subs(log(x), logx)
         s = self.args[0]._eval_nseries(x, n=n, logx=logx)
-        return (sign(direction)*s).expand()
+        if direction.is_zero:
+            return (s*sign(direction)).expand()
```

Comment:
We could just return `S.Zero` here. I think so this would be enough for now. I am not sure if the `is_extended_positive` or `is_extended_negative` methods should be use as direction can end up being complex (`I` or something). Do you see any case where the `ValueError` plays a role.

Reply:
cc @smichr

Reply:
If `s` is finite, 0 would be fine. If not, then it should be unevaluated, shouldn't it?

Reply:
Specifically while computing limits, I think the cases where `s` might not be finite are handled by the following lines inside the `.doit()` method of the `Limit` class:
```
if e.has(*_illegal):
    return self
```

To look at the general behavior of the _eval_nseries() method of the Abs class for cases where the variable `s` is infinite, I tried observing the following example where `s` assumes the value of `x + oo` :
```
x = Symbol('x')
k = S.Infinity
expr = Abs(sqrt(k)**2+x)
print(expr._eval_nseries(x,10,None))
```
The output of the code is `x + oo`.
 
Further, the following example raises a `NotImplementedError`:
```
x = Symbol('x')
k = S.Infinity
expr = 1/Abs(sqrt(k)**2+x)
print(expr._eval_nseries(x,10,None))
```